algoliasearch/algoliasearch.go

```diff
@@ -172,6 +172,55 @@ type Client interface {
 	// also accepts extra RequestOptions.
 	MultipleQueriesWithRequestOptions(queries []IndexedQuery, strategy string, opts *RequestOptions) (res []MultipleQueryRes, err error)
 
+	// List all the cluster available for the application
+	ListClusters() (res []Cluster, err error)
+
+	// ListClustersWithRequestOptions is the same as ListClusters but it
+	// also accepts extra RequestOptions.
+	ListClustersWithRequestOptions(opts *RequestOptions) (res []Cluster, err error)
+
+	// List all user IDs across all clusters
+	ListUserIDs(page int, hitsPerPage int) (res listUserIDsRes, err error)
+
+	// ListUserIDsWithRequestOptions is the same as ListUserIDs but it
+	// also accepts extra RequestOptions.
+	ListUserIDsWithRequestOptions(page int, hitsPerPage int, opts *RequestOptions) (res listUserIDsRes, err error)
+
+	// Get a specific user ID
+	GetUserID(userID string) (res UserID, err error)
+
+	// GetUserIDWithRequestOptions is the same as GetUserID but it
+	// also accepts extra RequestOptions.
+	GetUserIDWithRequestOptions(userID string, opts *RequestOptions) (res UserID, err error)
+
+	// Assign a user ID to a cluster
+	AssignUserID(userID string, clusterName string) (res AssignUserIDRes, err error)
+
+	// AssignUserIDWithRequestOptions is the same as AssignUserID but it
+	// also accepts extra RequestOptions.
+	AssignUserIDWithRequestOptions(userID string, clusterName string, opts *RequestOptions) (res AssignUserIDRes, err error)
+
+	// Remove a user ID from a cluster
```

Comment:
Missing `dot` and same as for `GetUserID`, we should perhaps add something in the comment saying this method is related to MCM.